libs/providers/config-cat/src/lib/config-cat-provider.ts

```diff
@@ -1,36 +1,54 @@
 import {
   EvaluationContext,
   JsonValue,
+  OpenFeatureEventEmitter,
   ParseError,
   Provider,
+  ProviderEvents,
   ResolutionDetails,
   ResolutionReason,
   StandardResolutionReasons,
   TypeMismatchError,
 } from '@openfeature/js-sdk';
-import * as configcat from 'configcat-js';
-import { IConfigCatClient } from 'configcat-js';
-import { SettingValue } from 'configcat-common/lib/RolloutEvaluator';
+import { getClient, IConfigCatClient, IEvaluationDetails, SettingValue } from 'configcat-js';
 import { transformContext } from './context-transformer';
 
 export class ConfigCatProvider implements Provider {
   private client: IConfigCatClient;
+  public events = new OpenFeatureEventEmitter();
+
+  public metadata = {
+    name: ConfigCatProvider.name,
+  };
 
   private constructor(client: IConfigCatClient) {
     this.client = client;
   }
 
-  public static create(...params: Parameters<typeof configcat.getClient>) {
-    return new ConfigCatProvider(configcat.getClient(...params));
+  public async initialize(context?: EvaluationContext): Promise<void> {
+    this.client.on('clientReady', () => this.events.emit(ProviderEvents.Ready));
```

Comment:
I think you want to change this so that you create a new Promise in this func, which is resolved with your client ready event, and return that.

Then the SDK will await it, and it will return ready itself.

Reply:
Oh totally overlooked that.
Committed a change.
I had to remove the option for a provider to be created by injecting a ConfigCat Client because there I could not safely assume to get the ClientReady event emitted. Only when subscribing to that event as a setupHook I can safely listen to that and await it.